api/src/main/java/org/apache/iceberg/transforms/Identity.java

```diff
@@ -38,6 +38,8 @@ class Identity<T> implements Transform<T, T> {
    */
   @Deprecated
   public static <I> Identity<I> get(Type type) {
+    Preconditions.checkArgument(!type.isVariantType(), "Unsupported type for identity: %s", type);
```

Comment:
don't we need to fix "canTransform" as well?

Reply:
Nvm I see that's already covered since "variant" is not considered a primitive

Reply:
Preconditions.checkArgument(type.typeId() != Types.VariantType);

Reply:
Since Variant is not a primitive type, canTransform() will be false.